engine-tests/src/test_utils/uniswap.rs

```diff
@@ -335,6 +345,42 @@ impl SwapRouter {
             data,
         }
     }
+
+    pub fn exact_input(&self, params: ExactInputParams, nonce: U256) -> TransactionLegacy {
+        let path: Vec<u8> = {
+            // The encoding here is 32-byte address, then 3-byte (24-bit) fee, alternating
+            let mut result = Vec::with_capacity(32 + 35 * params.path.len());
+            result.extend_from_slice(params.token_in.as_bytes());
```

Comment:
Just a nit, don't need to extend from slice here as it needlessly iterates. But its just a test, so I'm not to worried about it.